### ISSUE TITLE
Log environment info during staging phase

### ISSUE THREAD

**Issue by mrdavidlaing** [cloudfoundry-community/container-info-buildpack#1]
Gather info about the warden staging container, and write it to a file.

What info is useful?

**Comment by mrdavidlaing** [comment]
A good template? - https://gist.github.com/pmuellr/6925511

Script: https://github.com/pmuellr/cf-buildpack-dump/blob/master/bin/compile

**Comment by mrdavidlaing** [comment]
Resolved by 8c54e0f7cfb6a665126ed22a7dfae9e0b7151e62

The following is echoed during the staging phase:

```
-----> Describing compile environment
       -------------------------------------------------------------------------
       args: <build-dir>: /tmp/staged/app <cache-dir>: /tmp/cache
       -------------------------------------------------------------------------

       -------------------------------------------------------------------------
       env
       -------------------------------------------------------------------------
       USER=vcap
       VCAP_APPLICATION={"limits":{"mem":64,"disk":1024,"fds":16384},"application_version":"2f1dc4aa-79d6-4c89-b61a-9fbb93517074","application_name":"container-info-test","application_uris":["container-info-test.cfapps.io"],"version":"2f1dc4aa-79d6-4c89-b61a-9fbb93517074","name":"container-info-test","uris":["container-info-test.cfapps.io"],"users":null}
       STAGING_TIMEOUT=900.0
       PATH=/bin:/usr/bin
       PWD=/tmp/staged
       VCAP_SERVICES={}
       SHLVL=3
       HOME=/home/vcap
       BUILDPACK_CACHE=/var/vcap/packages/buildpack_cache
       MEMORY_LIMIT=64m
       _=/usr/bin/env

       -------------------------------------------------------------------------
       BUILD_DIR: /tmp/staged/app
       | | |___app
       | | | |___manifest.yml
       -------------------------------------------------------------------------

       -------------------------------------------------------------------------
       CACHE_DIR: /tmp/cache
       | |___cache
       -------------------------------------------------------------------------

       -------------------------------------------------------------------------
       ls /var/vcap/packages/buildpack_cache (ie, env BUILDPACK_CACHE)
       -------------------------------------------------------------------------
       | | | |___buildpack_cache
       | | | | |___apache-tomcat-7.0.37.tar.gz
       | | | | |___apache-tomcat-7.0.40.tar.gz
       | | | | |___apache-tomcat-7.0.41.tar.gz
       | | | | |___auto-reconfiguration-0.6.6.jar
       | | | | |___auto-reconfiguration-0.6.7.jar
       | | | | |___auto-reconfiguration-0.6.8.jar
       | | | | |___bundler-1.3.0.pre.5.tgz
       | | | | |___bundler-1.3.1.tgz
       | | | | |___bundler-1.3.2.tgz
       | | | | |___java-buildpack
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Fauto-reconfiguration%2Fauto-reconfiguration-0.7.0.jar.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Fauto-reconfiguration%2Findex.yml.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Fgroovy%2Fgroovy-2.1.6.zip.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Fgroovy%2Findex.yml.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Fnew-relic%2Findex.yml.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Fnew-relic%2Fnew-relic-2.21.3.jar.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Fopenjdk%2Flucid%2Fx86_64%2Findex.yml.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Fopenjdk%2Flucid%2Fx86_64%2Fopenjdk-1.6.0_27.tar.gz.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Fopenjdk%2Flucid%2Fx86_64%2Fopenjdk-1.7.0_21.tar.gz.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Fopenjdk%2Flucid%2Fx86_64%2Fopenjdk-1.8.0_M7.tar.gz.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Fplay-jpa-plugin%2Findex.yml.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Fplay-jpa-plugin%2Fplay-jpa-plugin-0.7.1.jar.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Fspring-boot-cli%2Findex.yml.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Fspring-boot-cli%2Fspring-boot-cli-0.5.0_M2.tar.gz.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Ftomcat%2Findex.yml.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Ftomcat%2Ftomcat-6.0.37.tar.gz.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Ftomcat%2Ftomcat-7.0.42.tar.gz.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Ftomcat-buildpack-support%2Findex.yml.cached
       | | | | | |___http:%2F%2Fdownload.pivotal.io.s3.amazonaws.com%2Ftomcat-buildpack-support%2Ftomcat-buildpack-support-1.1.1.jar.cached
       | | | | |___libyaml-0.1.4.tgz
       | | | | |___manifest.nodejs
       | | | | |___manifest.npm
       | | | | |___mysql-connector-java-5.1.12.jar
       | | | | |___node-0.4.7.tgz
       | | | | |___nodejs-0.10.0.tgz
       | | | | |___nodejs-0.4.10.tgz
       | | | | |___nodejs-0.4.12.tgz
       | | | | |___nodejs-0.4.7.tgz
       | | | | |___nodejs-0.6.20.tgz
       | | | | |___nodejs-0.8.21.tgz
       | | | | |___npm-1.0.106.tgz
       | | | | |___npm-1.1.65.tgz
       | | | | |___npm-1.2.12.tgz
       | | | | |___openjdk-1.6.0_27.tar.gz
       | | | | |___openjdk-1.7.0_21.tar.gz
       | | | | |___openjdk-1.7.0_25.tar.gz
       | | | | |___openjdk-1.8.0_M7.tar.gz
       | | | | |___play-jpa-plugin-0.6.6.jar
       | | | | |___postgresql-9.0-801.jdbc4.jar
       | | | | |___rails3_serve_static_assets.tgz
       | | | | |___rails_log_stdout.tgz
       | | | | |___ruby-1.8.7.tgz
       | | | | |___ruby-1.9.2.tgz
       | | | | |___ruby-1.9.3.tgz
       | | | | |___ruby-2.0.0.tgz
       | | | | |___ruby-build-1.8.7.tgz
       | | | | |___ruby-build-1.9.2.tgz
       | | | | |___ruby_versions.yml

       -------------------------------------------------------------------------
       Running processes
       wshd    
         `-bash
             `-ruby /var/vcap/packages/dea_next/buildpacks/bin/run ...
                 |-bash /tmp/buildpacks/container-info-buildpack.git/bin/compile /tmp/staged/app /tmp/cache
                 |   |-bash /tmp/buildpacks/container-info-buildpack.git/bin/describe_compile_env /tmp/staged/app /tmp/cache
                 |   |   `-pstree -a
                 |   `-bash /tmp/buildpacks/container-info-buildpack.git/bin/compile /tmp/staged/app /tmp/cache
                 |       `-sed -u s/^/       /
                 `-2*[{ruby}]
       -------------------------------------------------------------------------
```
